src/account/AccountList.js

```diff
@@ -24,9 +24,7 @@ export default function AccountList(props: Props) {
         renderItem={({ item, index }) => (
           <AccountItem
             index={index}
-            showDoneIcon={index === 0 && item.isLoggedIn}
-            email={item.email}
-            realm={item.realm}
+            accounts={item}
```

Comment:
I think there's enough pure refactoring of `AccountItem`'s props going on to motivate separating that work from the more interesting changes in the PR, on the [clear and coherent commits](https://zulip.readthedocs.io/en/latest/contributing/version-control.html) principle.

So, I'm thinking of at least one separate commit, for replacing the `email` and `realm` props with the `accounts` (should be singular, as mentioned elsewhere) prop.

And then maybe another commit for the `showDoneIcon` change, but I'm not sure that change is helpful; it puts responsibility on `AccountList` for knowing the details of the `state.accounts` data structure, which is something I think it wouldn't have to worry about, ideally. I think a better interface for `AccountList` than its current one, where it takes `index`, might be to have it take an `Identity` (see its jsdoc in src/types.js). But that improvement isn't necessary to get this feature in.

Then the more interesting changes would stand on their own and be easier to review and talk about in isolation.

Reply:
Pardon me, I considered showDoneIcon as a part of refactor, because if a component have all the resources then component itself should calculate what it's need